src/main/java/com/mraof/minestuck/entity/underling/UnderlingEntity.java

```diff
@@ -174,8 +173,15 @@ protected void onDeathUpdate()
 			{
 				for(GristAmount gristAmount : grist.getAmounts())
 				{
-					if(gristAmount.getAmount() > 0)
-						this.world.addEntity(new GristEntity(world, randX(), this.getPosY(), randZ(), gristAmount));
+					long gristLeft = gristAmount.getAmount();
+
+					while(gristLeft > (gristAmount.getAmount() / (int) ((Math.random() * 10) + 1)) && (long)(gristLeft*0.9) > 0) {
```

Comment:
I don't understand the purpose of the first condition in the while loop. I expect it to be some sort of condition for deciding if it should split the current amount of grist left, but can you give a more exact description?